server/index.js

```diff
@@ -123,7 +124,35 @@ app.get('/api/user/favorites', (req, res, next) => {
   });
 
 });
+app.get('/api/favorites', (req, res, next) => {
+
+  connection.query('SELECT * FROM favorites', (qErr, qRes) => {
+    if (qErr) {
+      console.error(qErr);
+      res.status(400).send('error retrieving venue');
+    } else {
+      const data = [];
+      qRes.forEach(row => {
+        data.push(JSON.parse(row.venue_json));
+      });
+      res.send(data);
+    }
+  });
+
+});
+app.get('/api/user/favorites/remove/:id', (req, res, next) => {
+  const id = req.params.id;
+  const query = "DELETE FROM favorites WHERE id = '" + id + "'";
+  connection.query(query, function (err) {
+    if (err) {
+
+      throw new ClientError(400, 'Information not found');
```

Comment:
You cannot throw inside of an asynchronous callback in Node.js. It works in `promise.then()`, like in `express-error-handling`, but not like this. it will just hang the client